smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -433,6 +435,77 @@ static void writeHostPrefix(GenerationContext context, OperationShape operation)
         });
     }
 
+    static void generateRpcEventStreamDeserializer(GenerationContext context,
+                                                   Shape target,
+                                                   String memberName,
+                                                   String assignTo) {
+        TypeScriptWriter writer = context.getWriter();
+        Symbol targetSymbol = context.getSymbolProvider().toSymbol(target);
+
+        writer.openBlock(assignTo + " = context.eventStreamMarshaller.deserialize(", ");", () -> {
+            writer.write("output.body,");
+            writer.openBlock("async event => {", "}", () -> {
+                writer.write("const eventName = Object.keys(event)[0];");
+                writer.openBlock("const parsedEvent = {", "};", () -> {
+                    writer.openBlock("$L: {", "}", memberName, () -> {
+                        writer.write("[eventName]: JSON.parse(Buffer.from(event[eventName].body).toString()),");
```

Comment:
This doesn't seem right. Can the event body be a blob(aka `eventPayload` trait)? In that case, `JSON.parse()` may not work here as it's for structured data. The event header trait is not handled here either. I don't think it's prohibited according to the [smithy doc](https://awslabs.github.io/smithy/1.0/spec/core/stream-traits.html?highlight=event%20stream#eventheader-trait), can you confirm that?

Reply:
I moved the parsing portion to JsonShapeDeserVisitor, and can check for streaming union members and RPC protocol there. This part now only adds in the marshaller usage.

The complexity of this is defeating me 😢, and I would like to limit the scope of the PR to the existing RPC deserialization case that fixes the Kinesis operation. 

If these changes don't make it harder to fully implement RPC eventstream later, I'd like to merge the fix in approximately this current state.